pkg/apis/security/v1alpha1/types.go

```diff
@@ -77,6 +81,9 @@ type Rule struct {
 	// destinations.
 	// +optional
 	To []NetworkPolicyPeer `json:"to"`
+	// RuleName describes the intention of this rule.
+	// +optional
+	RuleName string `json:"ruleName"`
```

Comment:
I think it could be `Name` as a field in `Rule` to keep align with other `Name` fields of K8s ones.
https://github.com/kubernetes/kubernetes/blob/e799c852fb337be414fa2dae982fed69a67806fe/pkg/apis/core/types.go#L2095
https://github.com/kubernetes/kubernetes/blob/e799c852fb337be414fa2dae982fed69a67806fe/pkg/apis/core/types.go#L3669
https://github.com/kubernetes/kubernetes/blob/e799c852fb337be414fa2dae982fed69a67806fe/pkg/apis/core/types.go#L48
And we should declare and verify the name is unique within the policy.

Reply:
Make sense. I can change it to `Name`.
Why do we need to make `Name` as unique? It seems that we only use the `Name` field to store extra info instead of using it as a key.

Reply:
We need it unique within a given policy resource so we can report stats information per rule instead of per policy (in an aggregated fashion)

Reply:
`Name` makes sense. for validation we will have to add additional checks in ValidationWebhook during CREATE and UPDATE